src/dsp_tools/commands/xmllib/different_approaches/factory_option.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+import warnings
+from dataclasses import dataclass
+from typing import Any
+
+from lxml import etree
+
+from dsp_tools.commands.xmllib.value_checkers import is_integer
+from dsp_tools.models.custom_warnings import DspToolsUserWarning
+
+"""
+Advantage: 
+    Creation and usage separated
+    Res ID is included in the warning
+Disadvantage: 
+    Value classes can be instantiated without check (otherwise we would get duplicate warnings)
+    Each value also needs a dedicated factory.
+"""
+
+
+class IntValueFactory:
```

Comment:
In its current form, I dislike this approach. But I think there is a way to make it better: 
 - I would not have a separate class `IntValueFactory`, but instead add the `create()` and `create_several()` methods as static methods to the `IntValue` class. 
 - Then, you can enforce the usage of the factory methods by setting all fields of the dataclass to `field(init=False)`. Like this, no one will be able to circumvent the factory.

Reply:
That I don't want. Because the `IntValue` is not responsible about creating itself. This is very unclean to me.

Reply:
Static factory methods are in my tool belt, so I just spent some time to do some research, to find out if I should remove it from there ;-) But it seems to be a legitimate pattern: 
 - https://stackoverflow.com/questions/929021
 - https://www.baeldung.com/java-constructors-vs-static-factory-methods

Reply:
I know it is, but I still dislike it because it doesn't separate the concerns cleanly to me.